src/components/GuestList/GuestList.tsx

```diff
@@ -45,33 +56,44 @@ const GuestList: React.FC<WeddingData> = ({
       value = value.slice(0,3) + "-" + value.slice(3,6) + "-" + value.slice(6);
     }
     setPhoneNumber(value)
-    setIsError(false)
+    setHasError(false)
     setErrorMessage('')
   }
 
   const submitGuest = (event: React.FormEvent) => {
     event.preventDefault();
+    setIsLoading(true)
+    const guestPost = {
+      name: guestName,
+      phoneNumber,
+      wedding: weddingId
+    }
     const newGuest: NewGuest = {
       id: Date.now(),
-      guestName,
+      name: guestName,
       phoneNumber
     }
-    if (guestName !== "" && phoneNumber.length === 12) {
-      setGuests([...guests, newGuest])
-      clearInputs();
-      setIsError(false)
+    if(guestName !== "" && phoneNumber.length === 12) {
+      clearInputs()
+      setHasError(false)
       setErrorMessage('')
-    } else if(guestName === "" && phoneNumber.length !== 12){
-      setIsError(true)
+      const postGuest = async () => {
+        await postAGuest(guestPost)
+      }
+      postGuest()
+      setGuests([...guests, newGuest])
+      updateGuests()
+    } else if (guestName === "" && phoneNumber.length !== 12) {
+      setHasError(true)
       setErrorMessage('Name and Phone Number Required')
-    } else if(guestName === "" ) {
-      setIsError(true)
+    } else if (guestName === "") {
+      setHasError(true)
       setErrorMessage('Name Required')
-    } else if(phoneNumber.length !== 12) {
-      setIsError(true)
+    } else if (phoneNumber.length !== 12) {
+      setHasError(true)
       setErrorMessage('Phone Number Required')
     }
-    // should be a POST request + adding card to UI
+    setIsLoading(false)
   }
```

Comment:
This submitGuest function is quite long, but I'm just happy it works! Refactoring can be a 'future iterations' task if we'd like